app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class ProcessMediaWikiJobsJob implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue, Queueable;
+
+    private string $wikiDomain;
+
+    public function __construct ( string $wikiDomain )
+    {
+        $this->wikiDomain = $wikiDomain;
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $mwPod = $kubernetesClient->pods()->setFieldSelector([
+            'status.phase' => 'Running'
+        ])->setLabelSelector([
+            'app.kubernetes.io/name' => 'mediawiki',
+            'app.kubernetes.io/component' => 'app-backend'
+        ])->first();
+
+        if (!$mwPod) {
+            $this->fail(
+                new \RuntimeException(
+                    'Unable to find a running MediaWiki pod in the cluster, '.
+                    'cannot continue.'
+                )
+            );
+            return;
+        }
+
+        $mwPod = $mwPod->toArray();
+
+        $jobSpec = new KubernetesJob([
+            'metadata' => [
+                'generateName' => 'run-all-mw-jobs-'
+            ],
+            'spec' => [
+                'ttlSecondsAfterFinished' => 60 * 60,
+                'template' => [
```

Comment:
I was wondering whether we should put these jobs into a dedicated namespace. In case yes, who should be responsible for creating the (empty) namespace? This job, or `wbaas-deploy`?

Reply:
I think yes we should probably want to keep stuff in a separate NS (see following comment too).

I suspect that responsibility for this would *probably* make sense to put in wbaas-deploy in some way rather than here. I would have expected it to be as simple as adding a Namespace yaml to the api chart.

Reply:
Ah, I somehow thought you couldn't yaml-spec a namespace, but it seems I was wrong: https://kubernetes.io/docs/tasks/administer-cluster/namespaces/ 

In that case, let's put it in the api chart, yes.